ocaml/xapi/xapi_vbd.ml

```diff
@@ -77,9 +77,14 @@ let plug ~__context ~self =
   let domid = Int64.to_int (Db.VM.get_domid ~__context ~self:vm) in
   let force_loopback_vbd = Helpers.force_loopback_vbd ~__context in
   let hvm = Helpers.has_qemu_currently ~__context ~self:vm in
-  if
-    System_domains.storage_driver_domain_of_vbd ~__context ~vbd:self = vm
-    && not force_loopback_vbd
+  if domid = 0
+  then (
+    (* SDS vm support *)
```

Comment:
You could say 'storage driver domain' support to make it clearer what this is

Reply:
Are you sure this is still needed though, the line below `storage_driver_domain_of_vbd` should give you the uuid of the driver domain, and then `vm` should be the uuid of dom0 when we plug the vbd into dom0, they should be different and it should take the `else` branch. If that is not happening do we have a bug in `storage_driver_domain_of_vbd`?

Reply:
Can you add some debug statements before the if to print the ref of the driver domain, and that of the vm?

Reply:
I agree with @edwintorok that, if the relevant PBD has the `storage_driver_domain_key` set in `other_config`, then this may not be necessary.